Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -68,13 +69,26 @@ namespace MultiplayerSample
 
             AZ_Error("EnergyBall", startSuccess, "Restart call for Energy Ball was unsuccessful.");
 
+            const Multiplayer::ConstNetworkEntityHandle cannonHandle = Multiplayer::GetNetworkEntityManager()->GetEntity(GetShooterNetEntityId());
+            if (cannonHandle.Exists())
+            {
+                const EnergyCannonComponent* cannonComponent = cannonHandle.FindComponent<EnergyCannonComponent>();
+                if (cannonComponent != nullptr)
+                {
+                    cannonComponent->KillBuildupEffect();
+                }
+            }
+
```

Comment:
Yeah, good call. I've kept the buildup termination logic contained within the cannon. It means we probably can't achieve frame-accurate buildup to fired energy ball transitions, but a frame or two of overlap either way probably won't be super noticeable.

Reply:
Oh - I guess what I was wondering was if the energy cannon could just listen to the same BallActiveChanged() event as the energy ball, so that they could still do it in the same frame. The RPC works too though.

Reply:
Ahhhh.. I understand.. since all the firing logic is server-side, the client-side version of the cannon has no idea about energy ball fire events at the moment unless the energy ball told the cannon 'hey, you just fired me'. So we'd need the RPC or the ball to cannon messaging either way.